Source/WebCore/layout/formattingContexts/inline/InlineContentBalancer.cpp

```diff
@@ -292,7 +292,7 @@ std::optional<Vector<LayoutUnit>> InlineContentBalancer::balanceRangeWithLineReq
             // Compute the cost of this line based on the line index
             for (size_t lineIndex = 1; lineIndex <= numberOfLines; lineIndex++) {
                 auto accumulatedCost = candidateLineCost + state[startIndex][lineIndex - 1].accumulatedCost;
-                if (accumulatedCost < state[breakIndex][lineIndex].accumulatedCost) {
+                if (accumulatedCost - LayoutUnit::epsilon() <= state[breakIndex][lineIndex].accumulatedCost) {
```

Comment:
what do these 2 floats have anything to do with LayoutUnit's precision value? is this about areEssentiallyEqual?

Reply:
yes, thanks! I wasn't aware of this function, but that's basically what we are trying to achieve. Let me try with that instead.